src/main/java/org/reactome/release/cosmicupdate/COSMICIdentifierUpdater.java

```diff
@@ -0,0 +1,194 @@
+	private boolean valid;
+	private Set<String> mutationIDs = new HashSet<>();
+	private String cosvIdentifier;
+
+	public String getIdentifier()
+	{
+		return identifier;
+	}
+	public void setIdentifier(String identifier)
+	{
+		this.identifier = identifier;
+	}
+	public Long getDbID()
+	{
+		return dbID;
+	}
+	public void setDbID(Long dbID)
+	{
+		this.dbID = dbID;
+	}
+	public String getSuggestedPrefix()
+	{
+		return suggestedPrefix;
+	}
+	public void setSuggestedPrefix(String suggestedPrefix)
+	{
+		this.suggestedPrefix = suggestedPrefix;
+	}
+	public boolean isValid()
+	{
+		return valid;
+	}
+	public void setValid(boolean valid)
+	{
+		this.valid = valid;
+	}
+	public Set<String> getMutationIDs()
+	{
+		return mutationIDs;
+	}
+	public void setMutationIDs(Set<String> mutationIDs)
+	{
+		this.mutationIDs = mutationIDs;
+	}
+	public String getCosvIdentifier()
+	{
+		return cosvIdentifier;
+	}
+	public void setCosvIdentifier(String cosvIdentifier)
+	{
+		this.cosvIdentifier = cosvIdentifier;
+	}
+	
+	@Override
+	public String toString()
+	{
+		return "[" + this.getDbID() + "; " 
+					+ this.getIdentifier()+ "; "
+					+ this.getSuggestedPrefix() + "; "
+					+ this.isValid() + "; "
+					+ this.getCosvIdentifier() + "; ("
+					+ this.getMutationIDs().stream().reduce("", (s, t) -> s + t + ",") + ")" + "]";
+	}
+	
+	/**
+	 * Implementation of compareTo ensures that these are sorted by validity, and then by identifier.
+	 * This is done so that the invalid identifiers are at the top of the report.
+	 */
+	@Override
+	public int compareTo(COSMICIdentifierUpdater other)
+	{
+		if (other.valid && !this.valid)
+		{
+			return -1;
+		}
+		if (!other.valid && this.valid)
+		{
+			return 1;
+		}
+		if (other.valid == this.valid)
+		{
+			return this.identifier.compareTo(other.identifier);
+		}
+		return 0;
+	}
+
+	/**
+	 * Perform an update of a COSMIC identifier.
+	 * @param adaptor - the database adapter to use.
+	 * @param creatorID - the DB_ID of the Creator of this update.
+	 * @throws Exception
+	 */
+	public void updateIdentfier(MySQLAdaptor adaptor, long creatorID) throws Exception
+	{
+		// If there is a COSV identifier, we'll update using that.
+		if (this.getCosvIdentifier() != null && !this.getCosvIdentifier().isEmpty())
+		{
+			// Create the InstanceEdit, if necessary.
+			synchronized (COSMICIdentifierUpdater.class)
+			{
+				if (COSMICIdentifierUpdater.updateWithNewCOSV == null)
+				{
+					COSMICIdentifierUpdater.updateWithNewCOSV = InstanceEditUtils.createDefaultIE(adaptor, creatorID, true, "Identifier was automatically updated to new \"COSV\" identifier by COSMIC Update process.");
+				}
+			}
+			GKInstance identifierObject = adaptor.fetchInstance( this.getDbID());
+			
+			updateIdentifierObject(adaptor, COSMICIdentifierUpdater.updateWithNewCOSV, identifierObject, this.getCosvIdentifier());
+		}
+		// If no COSV identifier was found, update using the suggested prefix (determined computationally).
```

Comment:
Could this comment have examples of the legacy prefixes? If its a long list of examples, then at least a couple would be good

Reply:
"COSM" is the legacy prefix.

Reply:
https://github.com/reactome/release-cosmic-update/blob/feature/initial-commits/src/main/java/org/reactome/release/cosmicupdate/COSMICUpdateUtil.java#L34

Reply:
But also, some identifiers in Reactome have _no_ prefix, they are just numeric. Which is what this code is trying to fix.